src/main/java/ch/njol/skript/expressions/ExprItemThrower.java

```diff
@@ -0,0 +1,78 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Item;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+@Name("Item Thrower")
+@Description("The player that thrown the dropped item.")
+@Examples({
+	"broadcast the item thrower of all dropped items"
+})
+@Since("INSERT VERSION")
+public class ExprItemThrower extends SimplePropertyExpression<Item, Object> {
+
+	static {
+		registerDefault(ExprItemThrower.class, Object.class, "[dropped] item thrower", "itementities");
+	}
+
+	@Override
+	public @Nullable Object convert(Item item) {
+		UUID uuid = item.getThrower();
+		if (uuid == null)
+			return null;
+		Entity checkEntity = Bukkit.getEntity(uuid);
+		if (checkEntity != null)
+			return checkEntity;
+		return Bukkit.getOfflinePlayer(uuid);
+	}
```

Comment:
Issue I have here is the method returns a UUID.
But this is forcing it to be an Entity (if its loaded) or a Player/OfflinePlayer.

Let's say for instance an entity dropped the item when it died.
This expression will return null when in actuality it's not technically null.
So if one were to do `... is set:` it will say false.

The same goes for the owner expression.

Reply:
That might be a good start. Also make sure to add that to the description if you do it that way.

Reply:
Alrighty